sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1718,22 +1718,26 @@ object PushPredicateThroughNonJoin extends Rule[LogicalPlan] with PredicateHelpe
     // implies that, for a given input row, the output are determined by the expression's initial
     // state and all the input rows processed before. In another word, the order of input rows
     // matters for non-deterministic expressions, while pushing down predicates changes the order.
-    // This also applies to Aggregate.
```

Comment:
what's the rationale for not applying this to Aggregate?

Reply:
When "Aggregate" contains both non-deterministic and deterministic expression fields, it can push down some deterministic filters. Because if the condition is deterministic, it certainly has no association with the non-deterministic expression field.

Reply:
For `Aggregate`, only filter which reference to groupExpression is pushed down. I can't think of what kind of Non-deterministic expressions will be affected by this push down.